rskj-core/src/test/java/co/rsk/db/HashMapBlocksIndex.java

```diff
@@ -67,4 +68,30 @@ public void flush() {
     public void close() {
 
     }
+
+    @Override
+    public void removeBlock(long blockNumber, Keccak256 blockHash) {
+        if (!index.containsKey(blockNumber)) {
+            return;
+        }
+
+        List<IndexedBlockStore.BlockInfo> blockInfoList = index.get(blockNumber);
+
+        if (blockInfoList == null) {
+            return;
+        }
+
+        List<IndexedBlockStore.BlockInfo> toRemove = new ArrayList<>();
+
+        for (IndexedBlockStore.BlockInfo bInfo : blockInfoList) {
+            if (bInfo.getHash().equals(blockHash)) {
+                toRemove.add(bInfo);
+            }
+        }
+        blockInfoList.removeAll(toRemove);
+        if (blockInfoList.isEmpty()) {
+            index.remove(blockNumber);
+        }
```

Comment:
In line with my comment in `MapDBBlocksIndex.java`, we can comment why here we don't need to do the put.